pkg/actions/store.go

```diff
@@ -0,0 +1,297 @@
+package actions
```

Comment:
I would still split implementations into 2 files so we can just delete the whole file when we finish the KV migration